test/e2e-tests-upgrade.sh

```diff
@@ -46,12 +46,15 @@ set +o pipefail
 header "Install the previous release of Tekton pipeline $PREVIOUS_PIPELINE_VERSION"
 install_pipeline_crd_version $PREVIOUS_PIPELINE_VERSION
 
+failed=0
+# The integration test for the older version to prevent regression on existing changes.
+go_test_e2e -timeout=20m ./test || failed=1
```

Comment:
It would be nice to update the initial comments (lines 21+) and describe what this script aims to test.

If I read this correctly, what you are doing here is to running the tests from `main` against the latest release of `Tekton`.
Unless tests are smart enough to skip themselves if a feature they rely on is not available, this test has a good chance of failing.

What we might consider doing is to run the tests from the latest release against `main`, which would ensure that no backward incompatible change was done.
Since there are cases where we do backward incompatible changes (in alpha and beta API versions, after the deprecation period), in these few cases we would have to call this out specifically in the tests, for instance we could have a mechanism to tell the tests to skip if the version of the target Tekton installation is greater than a certain value. 

This would still require to go backk and update the tests from the released versions though. An alternative could be to maintain in this script a version dependent exclusion list, to be passed to the `go test` command, with comments explaining the reason for the skip.

Reply:
Note that we don't necessarily need to run all the test in the old "released" version 👼🏼. We just need to run enough testing to make sure it works (because ideally it runs a released version, that should have had all its own test run and successful when we released), upgrade and run the `main` test on the upgraded instance to make sure it behave as intended.